### ISSUE TITLE
kafka-connector-hdfs not consuming json format data in kafka topic

### ISSUE THREAD

**Issue by phenixmzy** [confluentinc/kafka-connect-hdfs#208]
I used confluent-3.2.1(oss) version,want to pull json format data in kafka write to hdfs and specified in **partition.field.name** running on standalone mode.but it always not consuming kafka topic data.the log just show 
[2017-06-19 17:54:54,807] INFO Reflections took 19288 ms to scan 563 urls, producing 14078 keys and 89384 values  (org.reflections.Reflections:229)
what is problem, is my config have problem?

### connect-standalone.properties
bootstrap.servers=bigdata-32-201:9092,bigdata-32-202:9092,bigdata-32-203:9092

key.converter=org.apache.kafka.connect.json.JsonConverter
key.converter.schema.registry.url=http://bigdata-32-201:8081
value.converter=org.apache.kafka.connect.json.JsonConverter
value.converter.schema.registry.url=http://bigdata-32-201:8081
key.converter.schemas.enable=false
value.converter.schemas.enable=false

internal.key.converter=org.apache.kafka.connect.json.JsonConverter
internal.value.converter=org.apache.kafka.connect.json.JsonConverter
internal.key.converter.schemas.enable=false
internal.value.converter.schemas.enable=false

offset.storage.file.filename=/data/confluent/resourcemanager/connect.offsets

### quickstart-hdfs.properties
name=resource-consumer-hdfs-sink
connector.class=io.confluent.connect.hdfs.HdfsSinkConnector
tasks.max=1
topics=resource-consumer
hdfs.url=hdfs://mzcluster
hadoop.home=/opt/hadoop
hadoop.conf.dir=/opt/hadoop/etc/hadoop

format.class=io.confluent.connect.hdfs.parquet.ParquetFormat

partitioner.class=io.confluent.connect.hdfs.partitioner.FieldPartitioner
partition.field.name="end_time"

schema.compatibility=BACKWARD
rotate.interval.ms=3000
partition.duration.ms=100

flush.size=300

topics.dir=/confluent/resource-consumer/topics
logs.dir=/confluent/resource-consumer/logs

**Comment by phenixmzy** [comment]
# The all the log
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/opt/confluent-3.2.1/share/java/kafka-serde-tools/slf4j-log4j12-1.7.6.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/confluent-3.2.1/share/java/kafka-connect-elasticsearch/slf4j-simple-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/confluent-3.2.1/share/java/kafka-connect-hdfs/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/confluent-3.2.1/share/java/kafka-connect-s3/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/confluent-3.2.1/share/java/kafka-connect-storage-common/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/opt/confluent-3.2.1/share/java/kafka/slf4j-log4j12-1.7.21.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
[2017-06-19 17:54:34,722] INFO StandaloneConfig values: 
        access.control.allow.methods = 
        access.control.allow.origin = 
        bootstrap.servers = [bigdata-32-201:9092, bigdata-32-202:9092, bigdata-32-203:9092]
        internal.key.converter = class org.apache.kafka.connect.json.JsonConverter
        internal.value.converter = class org.apache.kafka.connect.json.JsonConverter
        key.converter = class org.apache.kafka.connect.json.JsonConverter
        offset.flush.interval.ms = 60000
        offset.flush.timeout.ms = 5000
        offset.storage.file.filename = /data/confluent/resourcemanager/connect.offsets
        rest.advertised.host.name = null
        rest.advertised.port = null
        rest.host.name = null
        rest.port = 8083
        task.shutdown.graceful.timeout.ms = 5000
        value.converter = class org.apache.kafka.connect.json.JsonConverter
 (org.apache.kafka.connect.runtime.standalone.StandaloneConfig:180)
[2017-06-19 17:54:34,945] INFO Logging initialized @850ms (org.eclipse.jetty.util.log:186)
[2017-06-19 17:54:35,401] INFO Kafka Connect starting (org.apache.kafka.connect.runtime.Connect:50)
[2017-06-19 17:54:35,402] INFO Herder starting (org.apache.kafka.connect.runtime.standalone.StandaloneHerder:71)
[2017-06-19 17:54:35,402] INFO Worker starting (org.apache.kafka.connect.runtime.Worker:119)
[2017-06-19 17:54:35,403] INFO Starting FileOffsetBackingStore with file /data/confluent/resourcemanager/connect.offsets (org.apache.kafka.connect.storage.FileOffsetBackingStore:60)
[2017-06-19 17:54:35,405] INFO Worker started (org.apache.kafka.connect.runtime.Worker:124)
[2017-06-19 17:54:35,406] INFO Herder started (org.apache.kafka.connect.runtime.standalone.StandaloneHerder:73)
[2017-06-19 17:54:35,406] INFO Starting REST server (org.apache.kafka.connect.runtime.rest.RestServer:98)
[2017-06-19 17:54:35,625] INFO jetty-9.2.15.v20160210 (org.eclipse.jetty.server.Server:327)
Jun 19, 2017 5:54:36 PM org.glassfish.jersey.internal.Errors logErrors
WARNING: The following warnings have been detected: WARNING: The (sub)resource method listConnectorPlugins in org.apache.kafka.connect.runtime.rest.resources.ConnectorPluginsResource contains empty path annotation.
WARNING: The (sub)resource method listConnectors in org.apache.kafka.connect.runtime.rest.resources.ConnectorsResource contains empty path annotation.
WARNING: The (sub)resource method createConnector in org.apache.kafka.connect.runtime.rest.resources.ConnectorsResource contains empty path annotation.
WARNING: The (sub)resource method serverInfo in org.apache.kafka.connect.runtime.rest.resources.RootResource contains empty path annotation.

[2017-06-19 17:54:36,698] INFO Started o.e.j.s.ServletContextHandler@19f7bd7b{/,null,AVAILABLE} (org.eclipse.jetty.server.handler.ContextHandler:744)
[2017-06-19 17:54:36,715] INFO Started ServerConnector@1fe0336a{HTTP/1.1}{0.0.0.0:8083} (org.eclipse.jetty.server.ServerConnector:266)
[2017-06-19 17:54:36,716] INFO Started @2623ms (org.eclipse.jetty.server.Server:379)
[2017-06-19 17:54:36,716] INFO REST server listening at http://172.17.32.201:8083/, advertising URL http://172.17.32.201:8083/ (org.apache.kafka.connect.runtime.rest.RestServer:150)
[2017-06-19 17:54:36,716] INFO Kafka Connect started (org.apache.kafka.connect.runtime.Connect:56)
[2017-06-19 17:54:36,738] INFO ConnectorConfig values: 
        connector.class = io.confluent.connect.hdfs.HdfsSinkConnector
        key.converter = null
        name = resource-consumer-hdfs-sink
        tasks.max = 1
        transforms = null
        value.converter = null
 (org.apache.kafka.connect.runtime.ConnectorConfig:180)
[2017-06-19 17:54:36,759] INFO Creating connector resource-consumer-hdfs-sink of type io.confluent.connect.hdfs.HdfsSinkConnector (org.apache.kafka.connect.runtime.Worker:178)
[2017-06-19 17:54:36,763] INFO Instantiated connector resource-consumer-hdfs-sink with version 3.2.1 of type class io.confluent.connect.hdfs.HdfsSinkConnector (org.apache.kafka.connect.runtime.Worker:181)
[2017-06-19 17:54:36,765] INFO HdfsSinkConnectorConfig values: 
        connect.hdfs.keytab = 
        connect.hdfs.principal = 
        filename.offset.zero.pad.width = 10
        flush.size = 300
        format.class = io.confluent.connect.hdfs.parquet.ParquetFormat
        hadoop.conf.dir = /opt/hadoop/etc/hadoop
        hadoop.home = /opt/hadoop
        hdfs.authentication.kerberos = false
        hdfs.namenode.principal = 
        hdfs.url = hdfs://mzcluster
        hive.conf.dir = 
        hive.database = default
        hive.home = 
        hive.integration = false
        hive.metastore.uris = 
        kerberos.ticket.renew.period.ms = 3600000
        locale = 
        logs.dir = /confluent/resource-consumer/logs
        partition.duration.ms = -1
        partition.field.name = "end_time"
        partitioner.class = io.confluent.connect.hdfs.partitioner.FieldPartitioner
        path.format = 
        retry.backoff.ms = 5000
        rotate.interval.ms = 3000
        rotate.schedule.interval.ms = -1
        schema.cache.size = 1000
        schema.compatibility = BACKWARD
        shutdown.timeout.ms = 3000
        storage.class = io.confluent.connect.hdfs.storage.HdfsStorage
        timezone = 
        topics.dir = /confluent/resource-consumer/topics
 (io.confluent.connect.hdfs.HdfsSinkConnectorConfig:180)
[2017-06-19 17:54:36,769] INFO Finished creating connector resource-consumer-hdfs-sink (org.apache.kafka.connect.runtime.Worker:194)
[2017-06-19 17:54:36,769] INFO SourceConnectorConfig values: 
        connector.class = io.confluent.connect.hdfs.HdfsSinkConnector
        key.converter = null
        name = resource-consumer-hdfs-sink
        tasks.max = 1
        transforms = null
        value.converter = null
 (org.apache.kafka.connect.runtime.SourceConnectorConfig:180)
[2017-06-19 17:54:36,773] INFO Creating task resource-consumer-hdfs-sink-0 (org.apache.kafka.connect.runtime.Worker:305)
[2017-06-19 17:54:36,773] INFO ConnectorConfig values: 
        connector.class = io.confluent.connect.hdfs.HdfsSinkConnector
        key.converter = null
        name = resource-consumer-hdfs-sink
        tasks.max = 1
        transforms = null
        value.converter = null
 (org.apache.kafka.connect.runtime.ConnectorConfig:180)
[2017-06-19 17:54:36,775] INFO TaskConfig values: 
        task.class = class io.confluent.connect.hdfs.HdfsSinkTask
 (org.apache.kafka.connect.runtime.TaskConfig:180)
[2017-06-19 17:54:36,775] INFO Instantiated task resource-consumer-hdfs-sink-0 with version 3.2.1 of type io.confluent.connect.hdfs.HdfsSinkTask (org.apache.kafka.connect.runtime.Worker:317)
[2017-06-19 17:54:36,808] INFO ConsumerConfig values: 
        auto.commit.interval.ms = 5000
        auto.offset.reset = earliest
        bootstrap.servers = [bigdata-32-201:9092, bigdata-32-202:9092, bigdata-32-203:9092]
        check.crcs = true
        client.id = 
        connections.max.idle.ms = 540000
        enable.auto.commit = false
        exclude.internal.topics = true
        fetch.max.bytes = 52428800
        fetch.max.wait.ms = 500
        fetch.min.bytes = 1
        group.id = connect-resource-consumer-hdfs-sink
        heartbeat.interval.ms = 3000
        interceptor.classes = null
        key.deserializer = class org.apache.kafka.common.serialization.ByteArrayDeserializer
        max.partition.fetch.bytes = 1048576
        max.poll.interval.ms = 300000
        max.poll.records = 500
        metadata.max.age.ms = 300000
        metric.reporters = []
        metrics.num.samples = 2
        metrics.recording.level = INFO
        metrics.sample.window.ms = 30000
        partition.assignment.strategy = [class org.apache.kafka.clients.consumer.RangeAssignor]
        receive.buffer.bytes = 65536
        reconnect.backoff.ms = 50
        request.timeout.ms = 305000
        retry.backoff.ms = 100
        sasl.jaas.config = null
        sasl.kerberos.kinit.cmd = /usr/bin/kinit
        sasl.kerberos.min.time.before.relogin = 60000
        sasl.kerberos.service.name = null
        sasl.kerberos.ticket.renew.jitter = 0.05
        sasl.kerberos.ticket.renew.window.factor = 0.8
        sasl.mechanism = GSSAPI
        security.protocol = PLAINTEXT
        send.buffer.bytes = 131072
        session.timeout.ms = 10000
        ssl.cipher.suites = null
        ssl.enabled.protocols = [TLSv1.2, TLSv1.1, TLSv1]
        ssl.endpoint.identification.algorithm = null
        ssl.key.password = null
        ssl.keymanager.algorithm = SunX509
        ssl.keystore.location = null
        ssl.keystore.password = null
        ssl.keystore.type = JKS
        ssl.protocol = TLS
        ssl.provider = null
        ssl.secure.random.implementation = null
        ssl.trustmanager.algorithm = PKIX
        ssl.truststore.location = null
        ssl.truststore.password = null
        ssl.truststore.type = JKS
        value.deserializer = class org.apache.kafka.common.serialization.ByteArrayDeserializer
 (org.apache.kafka.clients.consumer.ConsumerConfig:180)
[2017-06-19 17:54:36,934] INFO Kafka version : 0.10.2.1-cp1 (org.apache.kafka.common.utils.AppInfoParser:83)
[2017-06-19 17:54:36,934] INFO Kafka commitId : 078e7dc02a100018 (org.apache.kafka.common.utils.AppInfoParser:84)
[2017-06-19 17:54:36,938] INFO Created connector resource-consumer-hdfs-sink (org.apache.kafka.connect.cli.ConnectStandalone:90)
[2017-06-19 17:54:36,941] INFO HdfsSinkConnectorConfig values: 
        connect.hdfs.keytab = 
        connect.hdfs.principal = 
        filename.offset.zero.pad.width = 10
        flush.size = 300
        format.class = io.confluent.connect.hdfs.parquet.ParquetFormat
        hadoop.conf.dir = /opt/hadoop/etc/hadoop
        hadoop.home = /opt/hadoop
        hdfs.authentication.kerberos = false
        hdfs.namenode.principal = 
        hdfs.url = hdfs://mycluster
        hive.conf.dir = 
        hive.database = default
        hive.home = 
        hive.integration = false
        hive.metastore.uris = 
        kerberos.ticket.renew.period.ms = 3600000
        locale = 
        logs.dir = /confluent/resource-consumer/logs
        partition.duration.ms = -1
        partition.field.name = "end_time"
        partitioner.class = io.confluent.connect.hdfs.partitioner.FieldPartitioner
        path.format = 
        retry.backoff.ms = 5000
        rotate.interval.ms = 3000
        rotate.schedule.interval.ms = -1
        schema.cache.size = 1000
        schema.compatibility = BACKWARD
        shutdown.timeout.ms = 3000
        storage.class = io.confluent.connect.hdfs.storage.HdfsStorage
        timezone = 
        topics.dir = /confluent/resource-consumer/topics
 (io.confluent.connect.hdfs.HdfsSinkConnectorConfig:180)
[2017-06-19 17:54:37,173] INFO AvroDataConfig values: 
        enhanced.avro.schema.support = false
        schemas.cache.config = 1000
        connect.meta.data = true
 (io.confluent.connect.avro.AvroDataConfig:169)
[2017-06-19 17:54:37,178] INFO Hadoop configuration directory /opt/hadoop/etc/hadoop (io.confluent.connect.hdfs.DataWriter:93)
[2017-06-19 17:54:38,727] INFO Sink task WorkerSinkTask{id=resource-consumer-hdfs-sink-0} finished initialization and start (org.apache.kafka.connect.runtime.WorkerSinkTask:234)
[2017-06-19 17:54:38,841] INFO Discovered coordinator 172.17.32.203:9092 (id: 2147483645 rack: null) for group connect-resource-consumer-hdfs-sink. (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:586)
[2017-06-19 17:54:38,846] INFO Revoking previously assigned partitions [] for group connect-resource-consumer-hdfs-sink (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:397)
[2017-06-19 17:54:38,846] INFO (Re-)joining group connect-resource-consumer-hdfs-sink (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:420)
[2017-06-19 17:54:38,865] INFO Successfully joined group connect-resource-consumer-hdfs-sink with generation 1 (org.apache.kafka.clients.consumer.internals.AbstractCoordinator:388)
[2017-06-19 17:54:38,867] INFO Setting newly assigned partitions [resource-consumer-0] for group connect-resource-consumer-hdfs-sink (org.apache.kafka.clients.consumer.internals.ConsumerCoordinator:256)
[2017-06-19 17:54:38,888] INFO Started recovery for topic partition resource-consumer-0 (io.confluent.connect.hdfs.TopicPartitionWriter:208)
[2017-06-19 17:54:38,899] INFO Finished recovery for topic partition resource-consumer-0 (io.confluent.connect.hdfs.TopicPartitionWriter:223)
**[2017-06-19 17:54:54,807] INFO Reflections took 19288 ms to scan 563 urls, producing 14078 keys and 89384 values  (org.reflections.Reflections:229)**

**Comment by kkonstantine** [comment]
According to your configuration above you are attempting to use `JsonConverter` with schemas and you are setting the `key.converter.schema.registry.url` and `value.converter.schema.registry.url` properties. 

However, the `JsonConverter` is not integrated with Schema Registry. To inform Connect of schema-ed data while using the `JsonConverter` your records should include two top level fields named a) `schema` and b) `payload` (see the code for JsonConverter [here](https://github.com/apache/kafka/blob/trunk/connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverter.java))

Even in this case though, exporting such records with the HDFS connector is not currently possible (a [PR](https://github.com/confluentinc/kafka-connect-hdfs/pull/196) is in progress to allow this in one of the next releases). Currently it's only possible to export to HDFS JSON records that have been serialized with the `AvroConverter` when they were imported into Kafka. Then you can export such records as avro or parquet files using the equivalent format. 

I hope this helps.

**Comment by ewencp** [comment]
To add a bit more explanation, the HDFS connector did not start with any support for *schemaless* inputs, i.e. formats like JSON that load their data but do not have a fixed schema. The output formats the connector had supported all required schemas, and without any info from the input about the schema, there wasn't much we could do with the data.

We've been adding better support for schemaless data as long as the output format can handle the lack of a schema. However, as @kkonstantine points out some of this work is still in progress.

The `JsonConverter` does have a mode that allows you to include the schema inline in the envelope format that @kkonstantine describes and linked to. However, we would not recommend doing this in practice -- it is far too expensive and often times your schema will be significantly larger than the data itself... Rather, it was included in order to be able to use a single, popular Converter that could be used to demonstrate both schemaless and w/ schemas.

**Comment by kkonstantine** [comment]
Json file exports are in per https://github.com/confluentinc/kafka-connect-hdfs/pull/196. I believe we can close this.

**Comment by lalitakaushik** [comment]
@kkonstantine @ewencp If I am understanding it correctly, since this PR is merged now, we can use kafka connect to convert json records with schema to parquet in hdfs with the following configuration:

key.converter=org.apache.kafka.connect.json.JsonConverter
value.converter=org.apache.kafka.connect.json.JsonConverter
key.converter.schemas.enable=false
value.converter.schemas.enable=false

and 

format.class=io.confluent.connect.hdfs.parquet.ParquetFormat

and, my record will contain schema and payload embedded in it. one such example of record can be:
{"payload": {"f1":"teststring"}, "schema": {"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]}}

@ewencp as you mentioned you do not recommend this approach, so what should be the recommended way to achieve this.?

**Comment by ewencp** [comment]
@lalitakaushik Doing this with JSON is just generally going to be painful because JSON is schemaless. One option that doesn't have the overhead of including the schema in every message is to use a [single message transform](http://kafka.apache.org/documentation.html#connect_transforms) that adds schemas to schemaless data. For example, I started working on one that just [infers the schema based on the data](https://github.com/ewencp/kafka/commit/3abb54a8062fe727ddaabc4dd5a552dd0b465a03) and you could easily write one that adds/enforces a specific schema if you're not worried about schema evolution. These aren't provided out of the box, but you could grab that class and compile it into a jar yourself, then just add a transform something like

```
transforms=set-schema
transforms.set-schema.type=org.apache.kafka.connect.transforms.SetSchema.java
```

to your connector config.

**Comment by lalitakaushik** [comment]
@ewencp thanks.! I will try to work with your suggestion also. 
Can you also tell me if my understanding of the above mentioned approach is correct wrt configurations or if I am missing something. if correct, i will also try and run some tests to see what all overheads I am getting.

**Comment by ewencp** [comment]
@lalitakaushik The configs would only work with

```
key.converter.schemas.enable=true
value.converter.schemas.enable=true
```

since those enable inline schemas in the JSON data (i.e. the envelope format containing `schema` and `payload` fields). Otherwise it looks good.

**Comment by lalitakaushik** [comment]
Thanks @ewencp but Looks like I am facing some issue:
This is my connect distributed file:

key.converter=org.apache.kafka.connect.json.JsonConverter
value.converter=org.apache.kafka.connect.json.JsonConverter
key.converter.schemas.enable=true
value.converter.schemas.enable=true

key.converter.schema.registry.url=http://localhost:8081
value.converter.schema.registry.url=http://localhost:8081

internal.key.converter=org.apache.kafka.connect.json.JsonConverter
internal.value.converter=org.apache.kafka.connect.json.JsonConverter
internal.key.converter.schemas.enable=false
internal.value.converter.schemas.enable=false

and my format class is: ParquetFormat.

and this is the json data in my topic:
`./bin/kafka-console-producer --broker-list localhost:9092 --topic json-parquet-test
{"schema":{"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]},"payload":{"f1":"teststring1"}} `

I am getting this in the connect logs:

`[2017-11-28 14:54:57,900] ERROR Task json-parquet-test-0 threw an uncaught and unrecoverable exception (org.apache.kafka.connect.runtime.WorkerTask:142)
org.apache.kafka.connect.errors.DataException: JsonDeserializer with schemas.enable requires "schema" and "payload" fields and may not contain additional fields
	at org.apache.kafka.connect.json.JsonConverter.toConnectData(JsonConverter.java:332)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.convertMessages(WorkerSinkTask.java:345)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.poll(WorkerSinkTask.java:226)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.iteration(WorkerSinkTask.java:170)
	at org.apache.kafka.connect.runtime.WorkerSinkTask.execute(WorkerSinkTask.java:142)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:140)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:175)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
[2017-11-28 14:54:57,901] ERROR Task is being killed and will not recover until manually restarted (org.apache.kafka.connect.runtime.WorkerTask:143)`

Am I missing anything?
Let me know.

**Comment by ewencp** [comment]
You probably have data in the wrong format in your topic. The command you gave isn't how you use console producer -- you need to pass the records in via stdin. You can use console-consumer to check what data you have in your topic to see if there is some in the wrong format.

**Comment by lalitakaushik** [comment]
@ewencp I am using stdin itself:

`[ec2-user confluent-3.2.0]$ ./bin/kafka-console-producer --broker-list localhost:9092 --topic json-parquet-test5
{"payload": {"f1":"teststring"}, "schema": {"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]}}
{"payload": {"f1":"teststring1"}, "schema": {"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]}}`

and my console consumer also shows correctly:

`[ec2-user confluent-3.2.0]$ ./bin/kafka-console-consumer --bootstrap-server localhost:9092 --topic json-parquet-test5 --new-consumer --from-beginning
{"payload": {"f1":"teststring"}, "schema": {"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]}}
{"payload": {"f1":"teststring1"}, "schema": {"type":"record","name":"myrecord","fields":[{"name":"f1","type":"string"}]}}`

Anything else that I can do in it to unblock myself.? I am kind of stuck at it for the whole day.

**Comment by tony810430** [comment]
Hi @ewencp 
I used your configuration and sample message. After some modifications, I have uploaded parquet file successfully.
The modifications I made were:
* key.converter.schemas.enable=false (because the key is always null when I used console producer)
* schema definition `{"payload":{"f1":"teststring"},"schema":{"type":"struct","name":"mystruct","fields":[{"field":"f1","type":"string"}]}}
`. (It seems your sample message with a wrong schema definition, so I corrected it)
